server/utils/utils.py

```diff
@@ -0,0 +1,32 @@
+import contextlib
+import errno
+import socket
+
+def find_available_port(host, port):
+    """
+    Helper method to find open port on host. Tries 50 ports incremented from the specified port
+    """
+    new_port = port
+    found_port = False
+    num_ports_to_try = 50
+    with contextlib.closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
```

Comment:
I do not know if you can use the same socket and attempt to bind it multiple times.